testapi.pm

```diff
@@ -275,7 +275,12 @@ sub _check_backend_response {
             }
         }
         if (!$check && !$rsp->{saveresult}) {
-            OpenQA::Exception::FailedNeedle->throw(error => "no candidate needle with tag(s) '$mustmatch' matched", tags => $mustmatch);
+            # Must match can be only scalar or array ref.
+            local $" = ', ';
+            OpenQA::Exception::FailedNeedle->throw(
+                error => "no candidate needle with tag(s) '" . (ref($mustmatch) ? @$mustmatch : $mustmatch) . "' matched",
```

Comment:
I think it would be an ever better idea to join all elements when we have an array. https://perldoc.perl.org/functions/join.html should work